packages/filler/src/strategies/swap.ts

```diff
@@ -362,4 +405,126 @@ export class StableSwapFiller implements FillerStrategy {
 
 		return { calls, totalGasEstimate }
 	}
+
+	// Find whether uniswap v2 or v3 is the best protocol to use based on the amountIn the filler has to pay
```

Comment:
I'm not including V4 because it requires unlock patterns with `IUnlockCallback` that regular wallets can't handle directly. EIP-7702 workaround can be used, as the BatchExecutor contract code we are inheriting right now as an EOA can be updated. But it will add complexity to the contract. 

V2/V3 already provide good liquidity coverage with simpler integration, and V3 uses the same fee structure the V4 does.

Reply:
How does the callbackl affect swapping in the pool?

Reply:
Ohh, I see what you mean here https://docs.uniswap.org/contracts/v4/reference/core/PoolManager#unlock